cli/src/main/java/org/aya/cli/parse/AyaGKProducer.java

```diff
@@ -90,8 +90,11 @@ public record AyaGKProducer(
     if (node.is(IMPORT_CMD)) return ImmutableSeq.of(importCmd(node));
     if (node.is(MODULE)) return ImmutableSeq.of(module(node));
     if (node.is(OPEN_CMD)) return openCmd(node);
-    if (node.is(DECL)) {
-      var result = decl(node);
+    if (node.is(DECL) || node.peekChild(KW_PRIVATE) != null) {
```

Comment:
When do we get a `private something` but not a `private decl` ?
Are there any problems if we use `node.is(DECL)` ?

Reply:
`node.is(DECL)` is like `node instanceOf Decl` where `Decl` has 4 subclasses `Fn`, `Struct`, `Data` and `Prim`.
Similarly, `DECL` is a TokenSet of `FN`, `STRUCT`, `DATA` and `PRIM`

But GK doesn't parse `KW_PRIVATE? decl` to any node of type `FN`, `STRUCT`, `DATA` or `PRIM`, whenever the `KW_PRIVATE` is present and matched, instead, it produces a `DECL`-node.

We of course could add `AyaPsiElementTypes.DECL` to the `DECL` TokenSet, but we will need an extra `node = isPrivate ? node.child(DECL) : node` in either way.

Reply:
See the new proposal #690 which may address this issue in a more mature way.

Reply:
I see, thanks for your explanation.

Reply:
You're right! I am moving the private check to `decl`'s producer.